resources/assets/js/components/H2Components/Dialog.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from "react";
+import {
+  h2ComponentDialogLoad,
+  h2ComponentDialogLoadResize,
+  h2ComponentDialogEnableTrigger,
+} from "@hydrogen-design-system/system/dist/import/latest/components/dialog/scripts/dialog";
+import { defineMessages, useIntl } from "react-intl";
+import Button, { ButtonProps } from "./Button";
+
+const messages = defineMessages({
+  closeDialog: {
+    id: "hydrogen.dialog.close",
+    defaultMessage: "Close",
+  },
+});
+
+interface DialogContext {
+  id?: string;
+}
+
+const DialogContext = React.createContext<DialogContext>({});
+
+const useDialog = (): Partial<DialogContext> => {
+  const context = React.useContext(DialogContext);
+  if (!context) {
+    throw new Error("This component must be used within a <Dialog> component.");
+  }
+  return context;
+};
+
+const Actions: React.FunctionComponent = (props) => {
+  const { id = "dialog" } = useDialog();
+  const { children } = props;
+  return (
+    <div data-h2-dialog-actions {...props}>
+      {children}
+    </div>
+  );
+};
+
+const ActionBtn: React.FunctionComponent<ButtonProps> = (props) => {
+  const { id = "dialog" } = useDialog();
+  const { onClick, styling, type, children } = props;
+  const ref = React.useRef(null);
+  React.useEffect((): void => {
+    h2ComponentDialogEnableTrigger("latest", ref.current);
+  });
+  return (
+    <button
+      data-h2-button={styling}
+      ref={ref}
+      data-h2-dialog-trigger={`${id}`}
+      type={type}
+      onClick={onClick}
+      {...props}
+    >
+      {children}
+    </button>
+  );
+};
+
+const Content: React.FunctionComponent = (props) => {
+  const { id = "dialog" } = useDialog();
+  const { children } = props;
+  return (
+    <div data-h2-dialog-content id={`${id}Content`} {...props}>
+      {children}
+    </div>
+  );
+};
+const ExitBtn: React.FunctionComponent = (props) => {
+  useDialog();
+  const intl = useIntl();
+  const { children } = props;
+  return (
+    <Button data-h2-dialog-exit-trigger data-h2-bg-color="b(white, 0)">
+      <i className="fas fa-times" aria-hidden="true" />
+      <span data-h2-visibility="hidden">
+        {intl.formatMessage(messages.closeDialog)}
+      </span>
+    </Button>
```

Comment:
This component seems much less general than the others, and doesn't actually use `children` anywhere. Instead of including text, shouldn't that come from `children`?